specifyweb/backend/businessrules/uniqueness_rules.py

```diff
@@ -136,13 +135,38 @@ def get_exception(conflicts, matchable, field_map):
         if len(matchable.keys()) == 0 or set(all_fields) != set(field_map.keys()):
             continue
 
-        conflicts = model.objects.only('id').filter(**matchable)
+      
+        is_name_field = lambda field_path: field_path.split('__')[-1].lower() == 'name'
+
+        conflicts = model.objects.only('id')
+        exact_filters: dict[str, Any] = {}
+        binary_where_clauses: list[str] = []
+        binary_params: list[Any] = []
+        username_case_insensitive_q = Q()
+
+        for field_path, value in matchable.items():
+            if isinstance(value, str):
+                if model_name == "Specifyuser" and is_name_field(field_path):
+                    username_case_insensitive_q &= Q(**{f"{field_path}__iexact": value})
+                else:
+                    binary_where_clauses.append(f"BINARY {field_path} = %s")
+                    binary_params.append(value)
+            else:
+                exact_filters[field_path] = value
+
+        if exact_filters:
+            conflicts = conflicts.filter(**exact_filters)
+        if binary_where_clauses:
+            conflicts = conflicts.extra(where=binary_where_clauses, params=binary_params)
+        if username_case_insensitive_q:
+            conflicts = conflicts.filter(username_case_insensitive_q)
```

Comment:
(optional) 

To make this already long function more readable, can you extract this case-sensitivity logic into one or more separate functions? 

Just for example, perhaps all of this case-sensitivity can be collapsed into something like the following for  the `validate_unique` function: 

```py
conflicts = model.objects.only('id').filter(**case_insensitive_filters(matchable))
```
or 

```py
conflicts = case_insensitive_filter(model.objects.only('id'), matchable)
```